ratpack-pac4j/src/main/java/ratpack/pac4j/internal/Pac4jAuthenticationHandler.java

```diff
@@ -76,4 +78,9 @@ private void initiateAuthentication(final Context context) {
       }
     }).then(ignored -> webContext.sendResponse());
   }
+
+  private boolean isAjaxRequest(Request request) {
+    Headers headers = request.getHeaders();
+    return "XMLHttpRequest".equals(headers.get(HttpHeaderConstants.X_REQUESTED_WITH));
```

Comment:
Yeah, you're probably right.  I'm always hesitant to make hard use of non-standard HTTP headers, but this one is probably a case where the value is worth it.